### PR TITLE
fix: Properly overload `import.meta.env` with WXT's own environment globals

### LINKED ISSUE #935: WXT specific `import.meta.env` constants not showing up in editor
https://discord.com/channels/1212416027611365476/1224507127863967784/1277527012726145156

### PR DESCRIPTION
**Pull request by 1natsu172**
fix #935

I copied vite's implement style https://github.com/vitejs/vite/blob/main/packages/vite/types/importMeta.d.ts. It seems that module augmentation is not allowed because `import` is a top-level scope.

TS docs:
> You can’t declare new top-level declarations in the augmentation — just patches to existing declarations.

**completion screenshot**
image

### CODE REVIEW

File: packages/wxt/vitest.globalSetup.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  const e2eDistPath = './e2e/dist/';
+  if (existsSync(e2eDistPath)) {
+    try {
+      await rm(e2eDistPath, { recursive: true, force: true });
+      consola.info(`cleaned up ${e2eDistPath}`);
+    } catch (error) {
+      consola.error(error);
+    }
+  }
+}
```
Comment (1natsu172) [inline packages/wxt/vitest.globalSetup.ts:27]: The globalSetup and teardown was prepared as a side note. `e2e/dist` had swelled up to nearly GB.

File: packages/wxt/src/core/utils/building/generate-wxt-dir.ts [hunk 2]
```diff
@@ -201,7 +201,6 @@ async function getGlobalsDeclarationEntry(): Promise<WxtDirFileEntry> {
     path: 'types/globals.d.ts',
     text: [
       '// Generated by wxt',
-      'export {}',
```
Comment (aklinker1) [inline packages/wxt/src/core/utils/building/generate-wxt-dir.ts:204]: Very interesting, I think I copied this from Nuxt? I had always questioned why they had the export there, but I remember it working at some point lol.

File: packages/wxt/vitest.globalSetup.ts (hunk 1 above)
Comment (aklinker1) [reply packages/wxt/vitest.globalSetup.ts:27]: Let's extract this change to a separate PR, but it's a good fix. I thought I was doing this already though :/ maybe I'm thinking of a different project lol

**Review by aklinker1 — CHANGES_REQUESTED** [review]

**Comment by aklinker1** [comment]
Merged this PR with `main` to fix the checks as well.

File: packages/wxt/vitest.globalSetup.ts (hunk 1 above)
Comment (1natsu172) [reply packages/wxt/vitest.globalSetup.ts:27]: Haha common things. Okay, I'll separate this and submit!

**Comment by 1natsu172** [comment]
Haha, I missed the local rebase but it's maybe ok now 😅

**Review by aklinker1 — APPROVED** [review]

**Pull request merged**